weather_dl/download_pipeline/clients.py

```diff
@@ -152,6 +170,90 @@ def __exit__(self, exc_type, exc_value, traceback):
         self._redirector.__exit__(exc_type, exc_value, traceback)
 
 
+class APIRequestExtended(api.APIRequest):
```

Comment:
How about we include "MARS" in the name of this class?

Reply:
Naming discussion. I kind of like "extended", but is there a better name? What about "Adapter"? Any other ideas?

Reply:
On second thought, adding "MARS" to the name of the class may not be necessary.

Reply:
Named `SplitMARSRequest`.